mm2src/coins/eth.rs

```diff
@@ -784,6 +792,25 @@ impl EthCoinImpl {
         sha256(&input).to_vec()
     }
 
+    /// The id used to differentiate payments on Etomic swap smart contract.
+    /// For Taker Etomic Swap contract taker_secret_hash should be provided.
+    pub(crate) fn etomic_swap_v2_id(
+        &self,
+        funding_time_lock: u32,
+        payment_time_lock: u32,
+        secret_hash: &[u8],
+    ) -> Vec<u8> {
+        let funding_timelock_bytes = funding_time_lock.to_le_bytes();
+        let payment_timelock_bytes = payment_time_lock.to_le_bytes();
+
+        let mut input =
+            Vec::with_capacity(funding_timelock_bytes.len() + payment_timelock_bytes.len() + secret_hash.len());
+        input.extend_from_slice(&funding_timelock_bytes);
+        input.extend_from_slice(&payment_timelock_bytes);
+        input.extend_from_slice(secret_hash);
+        sha256(&input).to_vec()
```

Comment:
`extend_from_slice` is designed to clone elements from slice, while `extend` method extends a collection with the contents of an iterator. the `chain` method creates an iterator that links two or more iterators together.

According to manual and criterion benchmark tests (see the repo https://github.com/laruh/benchmarks_and_tests/tree/extend-from-slice-or-with-chain) both approaches yield the same result, and the performance difference is likely negligible in most practical scenarios.

`cargo bench` result
```
     Running benches/benchmark.rs (target/release/deps/benchmark-b2675adb094542c1)
Gnuplot not found, using plotters backend
extend_from_slice       time:   [76.041 ns 76.187 ns 76.343 ns]
Found 6 outliers among 100 measurements (6.00%)
  1 (1.00%) low mild
  3 (3.00%) high mild
  2 (2.00%) high severe

extend_with_chain       time:   [75.271 ns 75.359 ns 75.452 ns]
Found 5 outliers among 100 measurements (5.00%)
  4 (4.00%) high mild
  1 (1.00%) high severe
```
Time taken for functions running main
![Screenshot 2024-06-16 at 12 13 03](https://github.com/KomodoPlatform/komodo-defi-framework/assets/52405288/da9a17f6-288d-46b8-892f-c7fea5a31d1f)

I would like to provide my explanation of such benchmark results:
In both scenarios there will be allocation and elements cloning. 
`extend_from_slice` usage may reduce the number of allocation steps by combining all elements into a single iterator.
Yes, its true that `extend_from_slice` may involve more allocation steps because its called multiple times, but each call handles a contiguous slice which might have some performance optimizations.

I will change it to `extend` plus `chain` approach as it has slightly better performance

Reply:
thanks for the thorough research!

Reply:
Done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2129/commits/40ffb653942419305cfa95eeb71441faed3773d4

Reply:
did more iterations with `criterion`, both approaches performance difference is so negligible, they are identical i think
```
extend_from_slice       time:   [76.287 ns 76.535 ns 76.816 ns]                              
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild

extend_with_chain       time:   [76.303 ns 76.445 ns 76.586 ns]                              
Found 2 outliers among 100 measurements (2.00%)
  2 (2.00%) high mild
```

```
extend_from_slice       time:   [77.349 ns 77.527 ns 77.730 ns]                              
                        change: [+1.0600% +1.3658% +1.6707%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild

extend_with_chain       time:   [76.494 ns 76.648 ns 76.820 ns]                              
                        change: [+0.2788% +0.5390% +0.7899%] (p = 0.00 < 0.05)
                        Change within noise threshold.
```
  
```
extend_from_slice       time:   [76.307 ns 76.391 ns 76.486 ns]                              
                        change: [-1.8949% -1.6573% -1.4089%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 12 outliers among 100 measurements (12.00%)
  3 (3.00%) high mild
  9 (9.00%) high severe

extend_with_chain       time:   [76.303 ns 76.435 ns 76.561 ns]                              
                        change: [-0.8900% -0.5942% -0.2965%] (p = 0.00 < 0.05)
                        Change within noise threshold.
Found 6 outliers among 100 measurements (6.00%)
  2 (2.00%) low mild
  3 (3.00%) high mild
  1 (1.00%) high severe
```

```
extend_from_slice       time:   [76.953 ns 77.236 ns 77.529 ns]                              
                        change: [+0.2784% +0.5555% +0.8120%] (p = 0.00 < 0.05)
                        Change within noise threshold.
Found 8 outliers among 100 measurements (8.00%)
  3 (3.00%) high mild
  5 (5.00%) high severe

extend_with_chain       time:   [75.851 ns 75.991 ns 76.137 ns]                              
                        change: [-0.6184% -0.3356% -0.0483%] (p = 0.02 < 0.05)
                        Change within noise threshold.
Found 7 outliers among 100 measurements (7.00%)
  5 (5.00%) high mild
  2 (2.00%) high severe
```
```
extend_from_slice       time:   [76.969 ns 77.054 ns 77.146 ns]                              
                        change: [-1.1564% -0.9095% -0.6587%] (p = 0.00 < 0.05)
                        Change within noise threshold.
Found 14 outliers among 100 measurements (14.00%)
  6 (6.00%) high mild
  8 (8.00%) high severe

extend_with_chain       time:   [75.923 ns 76.087 ns 76.264 ns]                              
                        change: [-0.9150% -0.5842% -0.2325%] (p = 0.00 < 0.05)
                        Change within noise threshold.
Found 14 outliers among 100 measurements (14.00%)
  7 (7.00%) low mild
  2 (2.00%) high mild
  5 (5.00%) high severe
```

Reply:
seems so, what's the hyperfine command to run this and how many times did you tell it to run

Reply:
> seems so, what's the hyperfine command to run this and how many times did you tell it to run

I just run more iterations manually using the current criterion dependency. I will add hyperfine tool, but later, as such experiments and environment setup take some time :D